torch/_inductor/ir.py

```diff
@@ -5250,6 +5250,51 @@ def should_allocate(self):
         return False
 
 
+class TMADescriptor(ExternKernel):
+    def __init__(
+        self,
+        tensor: TensorBox,
+        dims: List[Union[int, torch.SymInt]],
+        block_dims: List[Union[int, torch.SymInt]],
+        element_size: Optional[Union[int, torch.SymInt]] = None,
```

Comment:
element_size should never be SymInt here

Reply:
I can change the type hint, but in theory it can be? `element_size` is passed as a separate argument to the `create_{1d,2d}_tma_descriptor` function in the user code. Although I agree that, in normal circumstances, it's taken from the underlying tensor's dtype byte size.